File: euler_024.py
from time import time
from itertools import permutations

start = time()

p = list(permutations([0,1,2,3,4,5,6,7,8,9],10))

n = "".join(map(str,p[999999]))

print("The 10^6th permutation: " + n)

stop = time()
print("Time: " + str(stop-start))

File: euler_024.js
const start = Date.now();

let factorials = [BigInt(1), BigInt(1)];
for (let i = 2; i < 10; i++) {
    factorials.push(BigInt(i) * factorials[i - 1]);
}
factorials.reverse();

let perm = "";
let rem = BigInt(1000000);
let digits = [...Array(10).keys()].map(x => BigInt(x));

for (const f of factorials) {
    let div = BigInt(0);
    while (rem > div * f) {
        div = div + BigInt(1);
    }
    div = div - BigInt(1);
    let digit = digits[Number(div)];    
    perm += digit;
    digits = digits.filter(x => x !== digit);
    rem -= div * f;
    console.log(rem);
}




const end = Date.now();

console.log("Execute time: ", end - start, " The millionth lexicographic permutation is: ", perm);